### PR TITLE
perf(api): new rate limits for OrganizationEventsEndpoint

### PR DESCRIPTION
**Pull request by vartec**
This PR 1 of 3, the other two: https://github.com/getsentry/getsentry/pull/13704 and  https://github.com/getsentry/sentry-options-automator/pull/1165

Rollout plan:
1. Add high-value, high traffic orgs  to list of orgs with increased limits (50 req / 1s)
2. Support notifies orgs that will be affected by reduced limit (1000 req / 5 min);
3. Rollout LA reduced limit;
4. Rollout EA reduced limit;
5. Consider if further limit reduction is needed;
6. Rollout GA reduced limit;
7. Make reduced limit new default and clean up reduced rollout flag;

Adds:
- `api.organization_events.rate-limit-increased.orgs` option, list of orgs with increased rate limit;
- `api.organization_events.rate-limit-increased.limits` option, dictionary allowing overriding default increased limit
- `organizations:api-organization_events-rate-limit-reduced-rollout` feature for LA/EA/GA roll-out of reduced 
- `api.organization_events.rate-limit-reduced.limits` option, dictionary allowing overriding default reduced limit

Decision tree showing which limit will be applied:
 ```mermaid
  flowchart TD
      A[Get organization] --> B{Organization\nexists}
      B -->|No| C[Return legacy rate limit]
      B -->|Yes| D{Organization\nin increased\nrate limit}
      D -->|Yes| E[Return increased rate limit]
      D -->|No| F{Organization in\nreduced limit\nroll-out}
      F -->|Yes| G[Return reduced rate limit]
      F -->|No| H[Return legacy rate limit]
  ```

### CODE REVIEW

File: src/sentry/api/endpoints/organization_events.py [hunk 1]
```diff
@@ -94,28 +93,65 @@
...
-
+LEGACY_RATE_LIMIT = dict(limit=30, window=1, concurrent_limit=15)
+# reduced limit will be the future default for all organizations not explicitly on increased limit
+DEFAULT_REDUCED_RATE_LIMIT = dict(
+    limit=1000, window=300, concurrent_limit=15  # 1000 requests per 5 minutes
+)
+DEFAULT_INCREASED_RATE_LIMIT = dict(limit=50, window=1, concurrent_limit=50)
+
+
+def rate_limit_events(
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/organization_events.py:104]: where this is called, have we not grabbed the organization yet? (i.e., is that why we're not passing in the org directly)?

File: src/sentry/api/endpoints/organization_events.py [hunk 2]
```diff
@@ -94,28 +93,65 @@
...
+
+    try:
+        organization = Organization.objects.get_from_cache(slug=organization_slug)
+    except Organization.DoesNotExist:
+        logger.warning("organization.slug.invalid", extra={"organization_slug": organization_slug})
+        return _config_for_limit(rate_limit)
+
+    if organization.id in options.get("api.organization_events.rate-limit-increased.orgs", []):
+        rate_limit = RateLimit(
+            **options.get(
```
Comment (JoshFerge) [inline src/sentry/api/endpoints/organization_events.py:140]: is it worth handling the case where someone incorrectly configures the option with invalid values? just thinking ahead for future people setting these -- how could that go wrong, if at all?

Comment (vartec) [reply src/sentry/api/endpoints/organization_events.py:140]: good point, I'll add handling that and falling back to defaults.

File: src/sentry/features/temporary.py [hunk 3]
```diff
@@ -47,6 +47,7 @@ def register_temporary_features(manager: FeatureManager):
     manager.add("organizations:anr-improvements", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
     manager.add("organizations:api-auth-provider", OrganizationFeature, FeatureHandlerStrategy.REMOTE)
     manager.add("organizations:api-keys", OrganizationFeature, FeatureHandlerStrategy.INTERNAL)
+    manager.add("organizations:api-organization_events-rate-limit-reduced-rollout", OrganizationFeature, FeatureHandlerStrategy.OPTIONS)
```
Comment (JoshFerge) [inline src/sentry/features/temporary.py:50]: is `OPTIONS` handler strategy ready to roll? have we tested it on a less high stakes option in prod?

File: src/sentry/api/endpoints/organization_events.py (hunk 2 above)
Comment (JoshFerge) [reply src/sentry/api/endpoints/organization_events.py:140]: maybe in the future we can have schema validation for options :D

File: src/sentry/api/endpoints/organization_events.py (hunk 1 above)
Comment (vartec) [reply src/sentry/api/endpoints/organization_events.py:104]: @JoshFerge  as far as I understand this is called _before_ the raw request parameters are converted to objects.

This is where it eventually gets called:

https://github.com/getsentry/sentry/blob/92e92bf00f5dc5abe5f48ab96fa61b87226bdb24/src/sentry/ratelimits/utils.py#L169-L170

Which in turn is called in `process_view(...)` of `RateLimitMiddleware`:
https://github.com/getsentry/sentry/blob/92e92bf00f5dc5abe5f48ab96fa61b87226bdb24/src/sentry/middleware/ratelimit.py#L65

File: src/sentry/features/temporary.py (hunk 3 above)
Comment (vartec) [reply src/sentry/features/temporary.py:50]: @wedamija how confident do you feel about this?

File: src/sentry/api/endpoints/organization_events.py (hunk 2 above)
Comment (vartec) [reply src/sentry/api/endpoints/organization_events.py]: I've added validation helper.

File: src/sentry/features/temporary.py (hunk 3 above)
Comment (vartec) [reply src/sentry/features/temporary.py:50]: going back to the old school way for now to minimize the risk

**Review by JoshFerge — APPROVED** [review]

**Pull request merged**